generated/nidmm/nidmm.proto

```diff
@@ -442,6 +470,12 @@ enum TransducerType {
   TRANSDUCER_TYPE_NIDMM_VAL_4_WIRE_RTD = 4;
 }
 
+enum TriggerDelays {
+  TRIGGER_DELAYS_UNSPECIFIED = 0;
+  TRIGGER_DELAYS_NIDMM_VAL_AUTO_DELAY_ON = -1;
```

Comment:
There is another VAL in the header file by the name `NIDMM_VAL_AUTO_DELAY`  with value same as `NIDMM_VAL_AUTO_DELAY_ON` (-1). So, should that be added as well on only these are fine? @reckenro

Reply:
Also, as per [protocol buffer documentation](https://developers.google.com/protocol-buffers/docs/proto#enum) negative values for enums are discouraged. Ryan, are we ok with negative values for enums?

> Enumerator constants must be in the range of a 32-bit integer. Since enum values use varint encoding on the wire, negative values are inefficient and thus not recommended.

Reply:
Yes. At the very least, we have a `-1` enum value in niscope's proto file as well (see [AcquisitionStatus](https://github.com/ni/grpc-device/blob/main/generated/niscope/niscope.proto#L332)).

I do think it might be worth re-evaluating if we support the `generate-mappings` tag for enums (see [this comment](https://github.com/doshirohan/grpc-device/pull/43#discussion_r622539539) in #43 for reference). If we support the `generate-mappings` tag for non-int32 enums it could make sense to also tag any enums with negative values to `generate-mappings` to avoid having negative enum values passed over the wire.

Reply:
About the negative enum values, I think we're ok with them. At the very least we weren't concerned about it for ni-scope. You can see we have a `-1` enum value in niscope's proto file (see [AcquisitionStatus](https://github.com/ni/grpc-device/blob/main/generated/niscope/niscope.proto#L332)).

I do think it might be worth re-evaluating if we end up supporting the `generate-mappings` tag for enums (see [this comment](https://github.com/doshirohan/grpc-device/pull/43#discussion_r622539539) in #43 for context). If we support the `generate-mappings` tag for non-int32 enums it could make sense to also tag any enums with negative values with it as well. That way, we can avoid having negative enum values passed over the wire.

Reply:
Don't include both auto delay values in the same enum. That would be confusing to anyone trying to figure out which of the two to use. If it's not in the base enums hapigen file in perforce, don't include it here.

Reply:
The [base metadata](https://p4.natinst.com/browser/perforce/Measurements/DMM/niDmmComponents/nidmmAPIMetadata/trunk/20.5/source/nidmmAPIMetadata/base/enums/enums.hapigen) doesn't seem to include "auto delay" enum itself. I guess we can go ahead with just ON and OFF enum values and drop the `NIDMM_VAL_AUTO_DELAY` to avoid confusion.

Reply:
Wait, if the trigger delays is not an enum in the base hapigen metadata, why are we making one here? I thought this change was simply adding things that were in base but omitted from nimi-python. I feel uneasy adding things here that aren't specific to gRPC but aren't in the base metadata. Adding grpc-specific overrides like naming, code-gen markers, etc. are okay, but changing metadata about the API itself is scary to me. We're eventually going to have to port any metadata changes we make back into the base hapigen metadata. If this is prompted by an expert in the driver team, then I'm okay with it. Otherwise we need to get their input on this kind of change.

Reply:
@reckenro, I was of the impression that we need to add values as enums if they are being used by certain functions and attributes. Even this step was suggested only for nimi python metadata?

Reply:
These are present in constants.hapigen in the base metadata folders, I will contact the driver experts to find out why they arent in the enums. We added them as enums because these values are being used by functions/attributes.

Reply:
Okay, the fact that they're coming from `constants.hapigen` improves the mater. Whoever gets tasked with creating a gRPC hapigen plugin will need to sort this out.